app/src/main/java/io/appium/uiautomator2/handler/GetSessionDetails.java

```diff
@@ -22,7 +23,14 @@ public GetSessionDetails(String mappedUri) {
         public AppiumResponse safeHandle(IHttpRequest request) {
             try {
                 JSONObject result = new JSONObject();
-                JSONObject lastScrollData = new JSONObject(AppiumUiAutomatorDriver.getInstance().getSession().getLastScrollData());
+                AccessibilityScrollData scrollData = AppiumUiAutomatorDriver.getInstance().getSession().getLastScrollData();
```

Comment:
We can move `session` to private field and simplify this code to:
```java
AccessibilityScrollData scrollData = session.getlastScrolldata();
if (scrollData != null) {
    result.put("lastScrollData", new JSONObject(scrollData.getAsmap()));
}
```

Reply:
aha, ok done :-)